dialogue-apache-hc5-client/src/main/java/com/palantir/dialogue/hc5/ApacheHttpClientChannels.java

```diff
@@ -425,14 +426,13 @@ public CloseableClient build() {
 
             setupConnectionPoolMetrics(conf.taggedMetricRegistry(), name, connectionManager);
 
+            ProxySelector configuredSelector = conf.proxy();
+            boolean noProxy = ProxySelector.getDefault().equals(configuredSelector);
```

Comment:
Ideally ClientConfigurations would provide a no-proxy selector instance by default, however that would break our workaround for the incorrect timeout being used with forward proxies if cjr was upgraded prior to dialogue. It can be safely done after the next hc5 release.